### ISSUE TITLE
Code Duplication in Cloud Plugins

### ISSUE THREAD

**Issue by kartg** [opensearch-project/OpenSearch#2155]
**Is your feature request related to a problem? Please describe.**
There seems to be duplication of logic in the code for Cloud Plugins. This is an opportunity for us to streamline the codebase and refactor these classes. This issue stems from [these comments](https://github.com/opensearch-project/OpenSearch/pull/2096#pullrequestreview-885245539) in #2096.

**Describe the solution you'd like**
Move common functionality to superclasses and have the Cloud Plugins inherit/extend these in platform-specific subclasses.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by willyborankin** [comment]
Small investigation I made.

Common code I found: 
List of common classes and functions for proxy settings in all cloud plugins:
- `ProxySettings` - could be a generic type, since all clients have they own `ProxyType` 
-  `*Settings` classes - the function `validateAndCreateProxySettings` has the same code for all plugins
Common functions:
- all `*Settings::getConfigValue` and `*Settings::getValue` classes use the same logic to get settings

Common settings:
- `client`
- `compress`
- `readonly`
- `chunk_size`
- `base_path`

I think it is possible to build high level abstraction for the plugin and re-use it in plugins.

**Comment by willyborankin** [comment]
@CEHENKLE and @kartg if you are still interested in refactoring cloud repositories. I can help with it. Wdyt?

**Comment by kartg** [comment]
@willyborankin we're definitely still interesting in refactoring these plugins. Would you be willing to attempt a first pass with one of the plugins, and we can iteratively refactor the rest?

**Comment by willyborankin** [comment]
Sure will prepare PR during this week

**Comment by sourav25998** [comment]
Hello @kartg . This issue seems to be a nice start to my open source contribution journey. Can you please assign this to me?

**Comment by dblock** [comment]
@sourav25998 i don’t think we can use assignments on GH due to permissions, but go for it!

**Comment by kartg** [comment]
@dblock I've noticed that sometimes it takes a comment to allow the assignment. I've assigned the issue to @sourav25998 now 😄

**Comment by Chaitanya-030** [comment]
Hii @kartg @dblock,
Is this issue still open? If yes, could you please assign it to me, would like to work on this.
